### PR TITLE
add inbox-as-contract.md

### PR DESCRIPTION
**Pull request by blockchaindevsh**

### CODE REVIEW

File: specs/experimental/inbox-as-contract.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+Currently the batch inbox is an EOA, this spec aims to allow the batch inbox to be a contract, to enable customized batch submission conditions like:
+1. The batch tx must be signed by a quorom of shared sequencers.
+2. The batch tx must pay DA fee to EthStorage to enable permanent and decentralized storage.
+
+## How It Works
+
+The integration mainly includes two parts:
+1. The [`BatchInboxAddress`](https://github.com/ethereum-optimism/optimism/blob/db107794c0b755bc38a8c62f11c49320c95c73db/op-chain-ops/genesis/config.go#L77) is replaced by an inbox contract to check submission conditions.
+2. When deriving, op-node should exclude failed batches.
```
Comment (qizhou) [inline specs/experimental/inbox-as-contract.md]:
```suggestion
2. When deriving, op-node should exclude failed batch txs.
```

File: specs/experimental/inbox-as-contract.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Inbox as Contract
+
+## Motivation
+
+Currently the batch inbox is an EOA, this spec aims to allow the batch inbox to be a contract, to enable customized batch submission conditions like:
```
Comment (qizhou) [inline specs/experimental/inbox-as-contract.md]:
```suggestion
Currently the batch inbox is an EOA.  (What are the pros and cons of EOA? How can we solve the cons?).  this spec aims to allow the batch inbox to be a contract, to enable customized batch submission conditions like:
```

File: specs/experimental/inbox-as-contract.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+1. The batch tx must be signed by a quorom of shared sequencers.
```
Comment (qizhou) [inline specs/experimental/inbox-as-contract.md]:
```suggestion
- The batch tx must be signed by a quorum of sequencers in a decentralized sequencing network; and/or
```

File: specs/experimental/inbox-as-contract.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+2. The batch tx must pay DA fee to EthStorage to enable permanent and decentralized storage.
```
Comment (qizhou) [inline specs/experimental/inbox-as-contract.md]:
```suggestion
- The batch tx must call a BLOB storage contract (e.g., EthStorage) with a DA fee, which is further distributed to data nodes that prove BLOB storage over time.
```

File: specs/experimental/inbox-as-contract.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+- Low submission gas cost due to the absence of onchain execution.
+- Verification logic is moved offchain to the derivation part, protected by a fault dispute game with a correct absolute prestate.
+
+Disadvantage:
+- Onchain verification is not possible.
+
+
+This specification aims to allow the batch inbox to be a contract, enabling customized batch submission conditions such as:
+- Requiring the batch transaction to be signed by a quorum of sequencers in a decentralized sequencing network; or
+- Mandating that the batch transaction call a BLOB storage contract (e.g., EthStorage) with a Data Availability (DA) fee, which is then distributed to data nodes that prove BLOB storage over time.
```
Comment (qizhou) [inline specs/experimental/inbox-as-contract.md]:
```suggestion
- Mandating that the batch transaction call a BLOB storage contract (e.g., EthStorage) with a long-term storage fee, which is then distributed to data nodes that prove BLOB storage over time.
```

File: specs/experimental/inbox-as-contract.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+## How It Works
+
+The integration process consists of two primary components:
+1. Replacement of the [`BatchInboxAddress`](https://github.com/ethereum-optimism/optimism/blob/db107794c0b755bc38a8c62f11c49320c95c73db/op-chain-ops/genesis/config.go#L77) with an inbox contract: The existing `BatchInboxAddress`, which currently points to an Externally Owned Account (EOA), will be replaced by a smart contract. This new inbox contract will be responsible for verifying and enforcing batch submission conditions.
+2. Modification of the op-node derivation process: The op-node will be updated to exclude failed batch transactions during the derivation process. This change ensures that only successfully executed batch transactions are processed and included in the derived state..
```
Comment (qizhou) [inline specs/experimental/inbox-as-contract.md]:
```suggestion
2. Modification of the op-node derivation process: The op-node will be updated to exclude failed batch transactions during the derivation process. This change ensures that only successfully executed batch transactions are processed and included in the derived state.
```

**Review by qizhou — APPROVED** [review]

**Comment by blockchaindevsh** [comment]
Close since the final version is on branch `feature/inbox_contract`.

**Pull request closed without merging**